### PR TITLE
#164069211 Configure Authors Haven to use PostgreSQL as DB Engine

### PR DESCRIPTION
**Pull request by joelethan**
#### What does this PR do?
- Configure Postgres to be the DB Engine 
#### Description of Task to be completed?
- Make modifications to use PostgreSQL as DB Engine
- Implement the use of Environment Variables to store Database Credentials
#### How should this be manually tested?
- Make sure PostgreSQL is installed and create a database.
- Clone the repo and cd into ah-backend-aquaman
    -  pip install -r requirements.txt
    -  python manage.py makemmigrations
    -  python manage.py migrate
    -  python manage.py createsuperuser
    - python manage.py runserver
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
- [#164069211](https://www.pivotaltracker.com/story/show/164069211)

### CODE REVIEW

**Comment by wasibani-roy** [comment]
@joelethan I still see the database as sqlite??

File: authors/settings.py [hunk 1]
```diff
@@ -81,8 +81,12 @@
...
 DATABASES = {
     'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': os.getenv("DB_NAME"),
+        'USER': os.getenv("USER_NAME"),
+        'PASSWORD': os.getenv("PASSWORD"),
+        'HOST': 'localhost',
+        'PORT': '5432'
```
Comment (CryceTruly) [inline authors/settings.py]: Good work @joelethan with securing key information with environment variables.
However please consider changing and hiding the DB service port as hackers look out for the default port.

**Comment by marthamareal** [comment]
Good work @joelethan , However you can consider making these changes to your PR.
- Add a sample of a `.env` file 
- Add postgres to the `requrements.txt` to ensure that its install
- update your PR title to follow the conventions in the playbook
- Attach reviewers on your PR by using the features on the sidebar of your PR(right-handside)
- Update your PR description basing on the playbook

File: authors/settings.py [hunk 2]
```diff
@@ -81,8 +81,12 @@
 
 DATABASES = {
     'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': os.getenv("DB_NAME"),
```
Comment (marthamareal) [inline authors/settings.py]: great work, I think you can add a default value to the function

File: authors/settings.py [hunk 3]
```diff
@@ -81,8 +81,12 @@
...
+        'USER': os.getenv("USER_NAME"),
+        'PASSWORD': os.getenv("PASSWORD"),
+        'HOST': 'localhost',
```
Comment (BagzieGracious) [inline authors/settings.py]: Good work, but try to hide the port and host. Make use of .env file

File: authors/settings.py (hunk 1 above)
Comment (joelethan) [reply authors/settings.py]: Thank you @CryceTruly, I have made the modifications. Awaiting more feedback now

File: requirements.txt [hunk 4]
```diff
@@ -2,5 +2,7 @@ Django==1.10
 django-cors-middleware==1.2.0
 django-extensions==1.7.1
 djangorestframework==3.4.4
+psycopg2==2.7.7
+psycopg2-binary==2.7.7
```
Comment (CryceTruly) [inline requirements.txt:28]: This is trivial but pyscopg2-binary makes psycopg2 obsolete so there should not be a need for use of both at the same time.

**Review by wasibani-roy — DISMISSED** [review]

**Review by Peace-Apple — DISMISSED** [review]
LGTM

**Review by Peace-Apple — DISMISSED** [review]

**Review by wasibani-roy — DISMISSED** [review]

**Review by CryceTruly — DISMISSED** [review]

**Review by BagzieGracious — DISMISSED** [review]

**Review by Peace-Apple — DISMISSED** [review]

**Review by wasibani-roy — DISMISSED** [review]

**Review by BagzieGracious — DISMISSED** [review]
LGTM

**Review by Peace-Apple — APPROVED** [review]

**Review by wasibani-roy — APPROVED** [review]

**Review by BagzieGracious — APPROVED** [review]
LGTM

**Pull request merged**